fossunited/foss_hackathon/doctype/foss_hackathon_team/test_foss_hackathon_team.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime, timedelta
+
+import frappe
+from faker import Faker
+from frappe.tests.utils import FrappeTestCase
+
+from fossunited.doctype_ids import HACKATHON, HACKATHON_PARTICIPANT, HACKATHON_TEAM
+
+
+class TestFOSSHackathonTeam(FrappeTestCase):
+    def setUp(self):
+        self.faker = Faker()
+        self.hackathon = frappe.get_doc(
+            {
+                "doctype": HACKATHON,
+                "hackathon_name": self.faker.name(),
+                "permalink": self.faker.slug(),
+                "start_date": datetime.now(),
+                "end_date": datetime.now() + timedelta(days=1),
+                "hackathon_description": self.faker.text(),
+                "max_team_size": 2,
+            }
+        ).insert()
+
+        self.team = frappe.get_doc(
+            {
+                "doctype": HACKATHON_TEAM,
+                "team_name": self.faker.name(),
+                "hackathon": self.hackathon.name,
+            }
+        ).insert()
+
+        self.participants = []
+        for _ in range(3):  # since max_team_size is 2, we need 3 participants to test the code
+            participant = frappe.get_doc(
+                {
+                    "doctype": HACKATHON_PARTICIPANT,
+                    "full_name": self.faker.name(),
+                    "email": self.faker.email(),
+                    "hackathon": self.hackathon.name,
+                }
+            ).insert()
+            self.participants.append(participant)
+
+    def tearDown(self):
+        self.team.delete()
+        for participant in self.participants:
+            participant.delete()
+        self.hackathon.delete()
+
+    def test_add_member_to_team(self):
+        for _ in range(self.hackathon.max_team_size):
+            self.team.append("members", {"member": self.participants[_].name})
+        self.team.save()
+        self.assertEqual(len(self.team.members), self.hackathon.max_team_size)
+
+    def test_add_member_exceeding_max_size(self):
+        for _ in range(self.hackathon.max_team_size + 1):
+            self.team.append("members", {"member": self.participants[_].name})
+        self.assertRaises(frappe.exceptions.ValidationError, self.team.save)
```

Comment:
shouldn't this be `with self.assertRaises()` ?

Reply:
`self.team.save()` right? the save function is not being called here

Reply:
Since CI is failing please test this in local with 

`bench --site <sitename> run-tests --app fossunited`

Reply:
I test it locally and it was working. self.team.save() is a callable and assertRaises can take a callable as it's second argument. 

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises

Reply:
@harshtandiya let me check on this again. 

the test is to check if the max team check actually works - when you mentioned you removed self.assertRaises line - it gets added and not saved - so it should show error, and when self.team.save() is kept alone error gets in. 

Actually: both 

```
with self.assertRaises(frappe.exceptions.ValidationError):
            self.team.save() 
```

and `self.assertRaises(frappe.exceptions.ValidationError, self.team.save)` should be doing the same thing but wrote differently

Reply:
Let me see if the original writing fails the test again

Reply:
To conclude @harshtandiya: self.team.save - won't do anything individually. You pass the function name into assertRaises and it gets executed.